cpp/src/arrow/csv/options.h

```diff
@@ -58,6 +61,8 @@ struct ARROW_EXPORT ParseOptions {
   /// Whether empty lines are ignored.  If false, an empty line represents
   /// a single empty value (assuming a one-column CSV file).
   bool ignore_empty_lines = true;
+  /// A handler function for rows which do not have the correct number of columns
+  util::optional<InvalidRowHandler> invalid_row_handler;
```

Comment:
Note that `util::optional` is probably not necessary: a `std::function` can be "empty": see https://en.cppreference.com/w/cpp/utility/functional/function/operator_bool

Reply:
Done